### PR TITLE
Feature/group requests

### PR DESCRIPTION
**Pull request by v3ceban**
Duplicate of #41

### CODE REVIEW

**Review by miaadele — APPROVED** [review]
Lots of great work, Vlad.

This is very nit-picky, just a small suggestion: instead of showing a "this project already has an advisor and co-advisor" pop up when a faculty member requests to join a project, maybe the button should be disabled instead? Not a big deal, though.

Note: For some reason, the nav links / routing seem to have been disrupted somehow....I switched my role in the prisma db to student, saved the change, yet I couldn't see the faculty directory.

**Comment by v3ceban** [comment]
> Note: For some reason, the nav links / routing seem to have been disrupted somehow....I switched my role in the prisma db to student, saved the change, yet I couldn't see the faculty directory.

It's because your role is saved in session. You need to re-login to update it

File: src/components/Requests.jsx
Comment (miaadele) [inline src/components/Requests.jsx:1]: For the demo, are we planning to leave the  "Advise this Project" and "Do NOT Advise this Project" buttons (both under the manage tab) how they are? Or will we be connecting a server-side action to the buttons?

**Comment by v3ceban** [comment]
> maybe the button should be disabled instead

will update later @miaadele

File: src/components/Requests.jsx
Comment (v3ceban) [reply src/components/Requests.jsx:1]: Good question....

Comment (miaadele) [reply src/components/Requests.jsx:1]: Let's discuss in class tomorrow :)

**Review by vaniaggarwal — APPROVED** [review]
It looks really good to me! Thank you Vlad, this was a lot of work

**Pull request merged**